.github/actions/neon-branch-create/action.yml

```diff
@@ -0,0 +1,162 @@
+      env:
+        ENVIRONMENT: ${{ inputs.environment }}
+        REGION_ID: ${{ inputs.region_id }}
+
+    - name: Get Parent Branch ID
+      id: parent-branch-id
+      shell: bash -euxo pipefail {0}
+      run: |
+        branches=$(curl "https://${API_HOST}/api/v2/projects/${PROJECT_ID}/branches" \
+          --fail \
+          --header "Accept: application/json" \
+          --header "Content-Type: application/json" \
+          --header "Authorization: Bearer ${API_KEY}"
+          )
+        parent_branch_id=$(echo $branches | jq --raw-output '.branches[0].id')
+        echo "parent_branch_id=${parent_branch_id}" >> $GITHUB_OUTPUT
+      env:
+        API_KEY: ${{ inputs.api_key }}
+        PROJECT_ID: ${{ inputs.project_id }}
+        API_HOST: ${{ steps.parse-input.outputs.api_host }}
+
+    - name: Create New Branch
+      id: create-branch
+      shell: bash -euxo pipefail {0}
+      run: |
+        branch=$(curl \
+          "https://${API_HOST}/api/v2/projects/${PROJECT_ID}/branches" \
+          --fail \
+          --header "Accept: application/json" \
+          --header "Content-Type: application/json" \
+          --header "Authorization: Bearer ${API_KEY}" \
+          --data "{
+            \"branch\": {
+              \"name\": \"Created by actions/neon-branch-create; GITHUB_RUN_ID=${GITHUB_RUN_ID} at $(date +%s)\",
+              \"parent_id\": \"${PARENT_BRANCH_ID}\"
+            }
+          }")
+
+        branch_id=$(echo $branch | jq --raw-output '.branch.id')
+        echo "branch_id=${branch_id}" >> $GITHUB_OUTPUT
+
+        endpoint_id=$(echo $branch | jq --raw-output '.endpoints[0].id')
+        echo "endpoint_id=${endpoint_id}" >> $GITHUB_OUTPUT
+      env:
+        API_KEY: ${{ inputs.api_key }}
+        API_HOST: ${{ steps.parse-input.outputs.api_host }}
+        PROJECT_ID: ${{ inputs.project_id }}
+        PARENT_BRANCH_ID: ${{ steps.parent-branch-id.outputs.parent_branch_id }}
+
+    - name: Get Role name
+      id: role-name
+      shell: bash -euxo pipefail {0}
+      run: |
+        roles=$(curl \
+          "https://${API_HOST}/api/v2/projects/${PROJECT_ID}/branches/${BRANCH_ID}/roles" \
+          --fail \
+          --header "Accept: application/json" \
+          --header "Content-Type: application/json" \
+          --header "Authorization: Bearer ${API_KEY}"
+          )
+
+        role_name=$(echo $roles | jq --raw-output '.roles[] | select(.name != "web_access") | .name')
+        echo "role_name=${role_name}" >> $GITHUB_OUTPUT
+      env:
+        API_KEY: ${{ inputs.api_key }}
+        API_HOST: ${{ steps.parse-input.outputs.api_host }}
+        PROJECT_ID: ${{ inputs.project_id }}
+        BRANCH_ID: ${{ steps.create-branch.outputs.branch_id }}
+
+    - name: Change Password
+      id: change-password
+      # A shell without `set -x` to not to expose password/dsn in logs
+      shell: bash -euo pipefail {0}
+      run: |
+        for i in $(seq 1 30); do
```

Comment:
I think 30 requests is too much. 3-5 is enough. 
Because it begins to work in a few requests or doesn't work at all :)

I'm just worried about load to the instances

Reply:
Reduced it to 10 (just to make it more than our typical compute start time, just in case mostly).